### ISSUE TITLE
А без джавы нельзя?  что там за формат?

### ISSUE THREAD

**Issue by socketpair** [miller7777777/LicDataExtractor#3]

**Comment by miller7777777** [comment]
Здравствуйте!

Без Java нельзя - данные вытаскивает из файла 1С-кий ring, а он на Java. LicDataExtractor - просто графическая обертка над ним.

Первая версия LicDataExtractor была вообще на PowerShell, потом я решил, раз все равно нужна Java, на ней же и сделать графическую оболочку.

Так что вопрос, скорее, к 1С.


>Вторник, 16 апреля 2019, 18:04 +03:00 от Коренберг Марк <notifications@github.com>:
>
>—
>You are receiving this because you are subscribed to this thread.
>Reply to this email directly,  view it on GitHub , or  mute the thread .

С уважением,
Игорь Рябинин
miller777@mail.ru

**Comment by socketpair** [comment]
Да, я там упоролся, поставил под линуксом этот ринг и жабу. и накатал следующее:

дебаг крайне важен. без него инфы сильно меньше.

```sh
#!/bin/bash

# apt-get install openjdk-11-jre-headless --no-install-recommends
# dpkg -i /opt/1C/v8.3/x86_64/ExtDst/*.deb

find /var/1C/licenses -type d | while read line; do
    /opt/1C/1CE/x86_64/ring/ring license list --path "$line" | while read pin_and_regnum; do
	/opt/1C/1CE/x86_64/ring/ring -l DEBUG license info --name $pin_and_regnum --path "$line" > $line/$pin_and_regnum.txt
    done
done
```

**Comment by socketpair** [comment]
Но тем не менее, раскурить бы сам формат. чтобы совсем без жабы. она меня бесит

**Comment by miller7777777** [comment]
Как вариант - написать в 1С. Зачем нужна именно Java, я не знаю. Нигде больше в 1С-ких продуктах java не используется, насколько я знаю. Выбор технологии выглядит странным, но зачем-то оно так сделано?

Ну и 1С явно могло бы сделать ring с GUI. Видимо, нет запроса от массы пользователей.
